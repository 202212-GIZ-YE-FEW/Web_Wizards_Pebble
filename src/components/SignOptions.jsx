import ButtonAuth from "./ButtonAuth";
import OrDivider from "./OrDivider";
import GoogleIcon from "../../public/GoogleIcon.png";
import TwitterIcon from "../../public/twitter.svg";

const SignOptions = ({ t }) => {
    return (
        <div className='flex flex-col-reverse order-last lg:flex-col lg:order-2'>
            <ButtonAuth
                text={t("signWithTwitter")}
                icon={TwitterIcon}
                alt='TwitterIcon'
                width='35'
                height='16.5'
            />
            <ButtonAuth
                text={t("signWithGoogle")}
                icon={GoogleIcon}
                alt='Google Icon'
                width='45'
            />
            <OrDivider text={t("or")} />
        </div>
    );
};

export default SignOptions;
