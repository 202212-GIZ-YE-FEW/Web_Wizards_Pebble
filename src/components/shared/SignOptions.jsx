import ButtonAuth from "@/components/shared/ButtonAuth";
import OrDivider from "@/components/shared/OrDivider";

import { signUpwithGoogle, singUpWithTwitter } from "@/firebase/auth";

import GoogleIcon from "~/GoogleIcon.png";
import TwitterIcon from "~/twitter.svg";
const SignOptions = ({ t }) => {
    return (
        <div className='flex flex-col-reverse order-last lg:flex-col lg:order-2'>
            <ButtonAuth
                onClick={singUpWithTwitter}
                text={t("signWithTwitter")}
                icon={TwitterIcon}
                alt='TwitterIcon'
                width='35'
                height='16.5'
            />
            <ButtonAuth
                onClick={signUpwithGoogle}
                text={t("signWithGoogle")}
                icon={GoogleIcon}
                alt='Google Icon'
                width='45'
            />
            <OrDivider text={t("or")} />
        </div>
    );
};

export default SignOptions;
