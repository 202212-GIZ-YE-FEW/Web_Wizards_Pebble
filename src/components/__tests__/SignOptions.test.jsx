import { I18nContext } from "next-i18next";
import renderer from "react-test-renderer";

import SignOptions from "../SignOptions";

it("renders correctly", () => {
    const i18n = {
        language: "en",
        t: (str) => str,
    };
    const tree = renderer
        .create(
            <I18nContext.Provider value={i18n}>
                <SignOptions t={i18n.t} />
            </I18nContext.Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
