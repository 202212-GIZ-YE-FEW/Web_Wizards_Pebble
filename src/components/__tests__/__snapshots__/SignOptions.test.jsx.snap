// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col-reverse order-last lg:flex-col lg:order-2"
>
  <button
    className="button__ButtonWrapper-sc-1aw98b4-0 button w-full h-14 px-6 transition"
    disabled={false}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid black",
        "borderRadius": "0.5rem",
        "boxShadow": "2px 2px #1a1a1a",
      }
    }
  >
    <div
      className="relative flex items-center space-x-4 justify-center"
    >
      <img
        alt="TwitterIcon"
        data-nimg="1"
        decoding="async"
        height={35}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={35}
      />
      <span
        className="block w-max font-medium tracking-wide text-gray-900 sm:text-base"
      >
        signWithTwitter
      </span>
    </div>
  </button>
  <button
    className="button__ButtonWrapper-sc-1aw98b4-0 button w-full h-14 px-6 transition"
    disabled={false}
    style={
      Object {
        "backgroundColor": "white",
        "border": "2px solid black",
        "borderRadius": "0.5rem",
        "boxShadow": "2px 2px #1a1a1a",
      }
    }
  >
    <div
      className="relative flex items-center space-x-4 justify-center"
    >
      <img
        alt="Google Icon"
        data-nimg="1"
        decoding="async"
        height={45}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={45}
      />
      <span
        className="block w-max font-medium tracking-wide text-gray-900 sm:text-base"
      >
        signWithGoogle
      </span>
    </div>
  </button>
  <div
    className="flex mt-7 order-3 items-center text-center mb-4"
  >
    <hr
      className=" text-gray-400  w-full "
    />
    <label
      className="block font-medium text-sm text-gray-700 ml-6 mr-6"
    >
      or
    </label>
    <hr
      className=" text-gray-400 w-full"
    />
  </div>
</div>
`;
